### PR TITLE
Endpoint to list models by revision

### LINKED ISSUE #776: Server: Add endpoint giving models it can serve for a given revision
Add a `GET` endpoint which takes in a project version/revision and lists the (names of the) models it has for that revision. Quite related to #775. Can be implemented in paralell, but we should agree on the final URL structure to make things reasonable.

### PR DESCRIPTION
**Pull request by milesgranger**
Will close #776

### CODE REVIEW

**Review by ryanjdillon — APPROVED** [review]
The test comment isn't that important, but I think it would be nice  :)

File: tests/gordo_components/server/test_gordo_server.py [hunk 1]
```diff
@@ -139,3 +139,45 @@ def listdir_fail(*args, **kwargs):
...
     assert resp.json["available-revisions"] == [expected_revision]
+
+
+@pytest.mark.parametrize(
+    "revision_to_models",
+    [
+        {"123": ("model-1", "model-2"), "456": ("model-3", "model-4")},
+        {"123": (), "456": ("model-1",)},
+        dict(),
+    ],
```
[the remark this answers is not in the document]
Comment (milesgranger) [reply tests/gordo_components/server/test_gordo_server.py:150]: Is this what you mean, something like this?
```python
@pytest.mark.parametrize(
    "revisions,models",
    [
        (("123", "456"), (("model-1", "model-2"), ("model-3", "model-4")))
    ],
)
```
I find that much harder to read. Maybe I misunderstand. The thing here is that there are multiple revisions and each revision can hold multiple models.

File: gordo_components/server/views/base.py [hunk 2]
```diff
@@ -273,7 +273,27 @@ def get(self, gordo_project: str):
...
 
 
+class ModelsByRevisionListView(Resource):
+    """
+    List the models which can be served for a given revision
+    """
+
+    @api.doc(description="Available models for a given revision.")
+    def get(self, gordo_project: str, revision: str):
+        collection_dir = os.environ[current_app.config["MODEL_COLLECTION_DIR_ENV_VAR"]]
```
[the remark this answers is not in the document]
Comment (milesgranger) [reply gordo_components/server/views/base.py:283]: It's just one of those things that have been there, I guess allowing us to change the name of the environment variable without changing code like this.. :man_shrugging:

File: tests/gordo_components/server/test_gordo_server.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (milesgranger) [reply tests/gordo_components/server/test_gordo_server.py:150]: Updated the test doc for maybe better clarification. :+1:

**Pull request merged**